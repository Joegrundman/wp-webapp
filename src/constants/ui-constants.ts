export const urlMapEur: string = 'http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg';
export const urlMapPac: string = 'http://res.cloudinary.com/druzhkwmt/image/upload/v1456179470/evnyvf9x1ehyxt3ksrkp.jpg';
