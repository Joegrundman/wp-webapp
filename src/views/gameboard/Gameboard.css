.main {
  height: 100%;
}

.mapContainer {
  /* height: 500px;
  width: 1300px; */
  overflow: scroll;
}

.mapEur {
  background: url(http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg);
}

.mapPac {
  background: url(http://res.cloudinary.com/druzhkwmt/image/upload/v1456179470/evnyvf9x1ehyxt3ksrkp.jpg);
}
