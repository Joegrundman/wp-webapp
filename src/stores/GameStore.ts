import { URL_MAP_EUR, URL_MAP_PAC } from 'Constants/ui-constants'
import { observable } from 'mobx'

class GameStore {

  @observable public theater: string = URL_MAP_EUR
  @observable public showDialogPanel: boolean = false
  @observable public currentActiveDialog: string = ''

  public setTheater (theater: string): void {
    this.theater = theater
  }

  public toggleTheater (): void {
    if (this.theater === URL_MAP_EUR) {
      this.theater = URL_MAP_PAC
    } else {
      this.theater = URL_MAP_EUR
    }
  }

  public toggleDialogPanel (): void {
    this.showDialogPanel = !this.showDialogPanel
  }

  public openDialogPanel (): void {
    this.showDialogPanel = true
  }

  public handleDialogButton(dialog: string): void {
    if(this.showDialogPanel && this.currentActiveDialog !== dialog) {
      this.setCurrentActiveDialog(dialog)
    } else if(this.showDialogPanel && this.currentActiveDialog === dialog) {
      this.closeDialogPanel()
      this.setCurrentActiveDialog('')
    } else {
      this.openDialogPanel()
      this.setCurrentActiveDialog(dialog)
    }  
  }

  public closeDialogPanel (): void {
    this.showDialogPanel = false
  }

  public setCurrentActiveDialog (dialog: string): void {
    this.currentActiveDialog = dialog
  }
}

export default GameStore