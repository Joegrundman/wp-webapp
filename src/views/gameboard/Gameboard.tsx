import MainMap, { IMapOpts } from 'Components/mainMap/MainMap'
import Game from 'Game/game'
import { initialize } from 'Init/init'
import { observer } from 'mobx-react'
import * as React from 'react'
import GameStore from 'Stores/GameStore'
import { handleKeyboardEvent, handleKeyupEvent } from 'Wp/Eventing/eventing-keyboard'
import { getGame } from 'Wp/Game'
import locals from './Gameboard.css'

interface IGameState {
  initialized: boolean
  windowWidth: number
  windowHeight: number
}

export interface IGameProps {
  store: GameStore
}

@observer
class GameBoard extends React.Component<IGameProps, IGameState> {
  public mainMapContainer: React.RefObject<HTMLDivElement>
  public game: Game
  public mapCanvas: HTMLCanvasElement
  public mapCtx: CanvasRenderingContext2D
  public doubleClickListener: EventListenerOrEventListenerObject
  public mouseDownListener: EventListenerOrEventListenerObject
  public mouseMoveListener: EventListenerOrEventListenerObject
  public mouseUpListener: EventListenerOrEventListenerObject
  public scrollListener: EventListenerOrEventListenerObject
  
  constructor(props: IGameProps) {
    super(props)

    this.state = {
      initialized: false,
      windowHeight: 0,
      windowWidth: 0,
    }
    this.mainMapContainer = React.createRef()
    this.updateWindowDimensions = this.updateWindowDimensions.bind(this)
  }

  public componentDidMount(): void {
    this.updateWindowDimensions()
    window.addEventListener('resize', this.updateWindowDimensions)
  }
  
  public componentWillUnmount(): void {
    window.removeEventListener('resize', this.updateWindowDimensions)
  }

  public attachKeyboardEvents(): void {
    window.addEventListener('keydown', (e: KeyboardEvent): void => handleKeyboardEvent(e.keyCode))
    window.addEventListener('keyup', (e: KeyboardEvent): void => handleKeyupEvent())
  }

  public attachMapEvents(): void {
    this.mouseDownListener = this.game.currentMap.onMouseDown.bind(this.game.currentMap)
    this.mouseMoveListener = this.game.currentMap.onMouseMove.bind(this.game.currentMap)
    this.mouseUpListener = this.game.currentMap.onMouseUp.bind(this.game.currentMap)
    this.scrollListener = this.game.currentMap.onScroll.bind(this.game.currentMap)
    this.doubleClickListener = this.game.currentMap.onDoubleClick.bind(this.game.currentMap)

    this.mapCanvas.addEventListener('mousemove', this.mouseMoveListener, false)
    this.mapCanvas.addEventListener('mousedown', this.mouseDownListener, false)
    this.mapCanvas.addEventListener('mouseup', this.mouseUpListener, false)
    this.mapCanvas.addEventListener('dblclick', this.doubleClickListener, false)
    if (this.mainMapContainer.current) {
      this.mainMapContainer.current.addEventListener('scroll', this.scrollListener, true)
    }
  }

  public removeMapEvents(): void {
    this.mapCanvas.removeEventListener('mousemove', this.mouseMoveListener, false)
    this.mapCanvas.removeEventListener('mousedown', this.mouseDownListener, false)
    this.mapCanvas.removeEventListener('mouseup', this.mouseUpListener, false)
    this.mapCanvas.removeEventListener('dblclick', this.doubleClickListener, false)
    if (this.mainMapContainer.current) {
      this.mainMapContainer.current.removeEventListener('scroll', this.scrollListener as EventListenerOrEventListenerObject, true)
    }
    this.game.currentMap.mapCanvas = this.mapCanvas
  }

  public getMapContext = (
    mapCanvas: HTMLCanvasElement,
    mapCtx: CanvasRenderingContext2D,
    opts: IMapOpts): void => {
    this.mapCanvas = mapCanvas
    this.mapCtx = mapCtx

    /**
     * Initialize the wp this.state.game instance and start the scripts
     */
    if (!this.state.initialized) {
      initialize(this.mapCtx, this.mapCanvas, opts)
      this.game = getGame()
      this.attachMapEvents()
      this.attachKeyboardEvents()
      this.setState({
        initialized: true
      })
    }
    else {
      this.game.switchTheaters()
      this.attachMapEvents()
    }
  }
  
  public toggleTheater = (): void => {
    this.removeMapEvents()
    this.props.store.toggleTheater()
  }

  public updateWindowDimensions(): void {
    this.setState({ windowWidth: window.innerWidth, windowHeight: window.innerHeight });
  }

  public render(): JSX.Element {

    return (
      <div className={locals.main}>
        <div ref={this.mainMapContainer} 
          style={{
            height: `${this.state.windowHeight - 90}px`,
            width: `${this.state.windowWidth - 60}px`
          }}
          className={locals.mapContainer}>
          <MainMap getMapContext={this.getMapContext} url={this.props.store.theater} />
        </div>
      </div>
    )
  }
}

export default GameBoard
