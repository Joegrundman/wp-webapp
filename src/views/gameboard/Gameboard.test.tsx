import { URL_MAP_PAC } from 'Constants/ui-constants'
import { shallow, ShallowWrapper } from 'enzyme'
import * as React from 'react'
import GameStore from 'Stores/GameStore'
import Gameboard, { IGameProps } from './Gameboard'

jest.mock('Wp/Game', () => ({
  getGame: () => ({
    switchTheaters: jest.fn()
  })
}))

jest.mock('Init/init', () => ({
  initialize: jest.fn()
}))


describe('<Gameboard />', (): void => {
  let store: GameStore
  let defaultProps: any

  const shallowComponent = (props: IGameProps): ShallowWrapper => shallow(<Gameboard {...props} />)

  beforeEach((): void => {
    store = new GameStore()
    defaultProps = {
      store
    }
  })

  describe('@render', (): void => {
    it('should render as expected in the default state', (): void => {
      const component: ShallowWrapper = shallowComponent(defaultProps)
      expect(component).toMatchSnapshot()
    })

    it('should render as expected with different theater', (): void => {
      store.toggleTheater()
      const component: ShallowWrapper = shallowComponent(defaultProps)
      expect(component).toMatchSnapshot()
    })

    it('should render as expected with canvas dimensions set on the state', (): void => {
      const component: ShallowWrapper = shallowComponent(defaultProps)
      component.setState({
        windowHeight: 300,
        windowWidth: 500,
      })
      expect(component).toMatchSnapshot()
    })
  })

  describe('@lifecycle', (): void => {
    describe('componentDidMount', (): void => {
      it('should call updateWindowDimensions and set a resize eventlistener', () => {
        const mockEvent: jest.SpyInstance<any> = jest.spyOn(window, 'addEventListener')
        const component: ShallowWrapper = shallowComponent(defaultProps)
        const instance: Gameboard = component.instance() as Gameboard
        instance.updateWindowDimensions = jest.fn()
        instance.componentDidMount()
        expect(instance.updateWindowDimensions).toHaveBeenCalled()
        expect(mockEvent.mock.calls[0][0]).toBe('resize')
      })
    })

    describe('componentWillUnmount', (): void => {
      it('should remove the resize event listener', (): void => {
        const mockEvent: jest.SpyInstance<any> = jest.spyOn(window, 'removeEventListener')
        const component: ShallowWrapper = shallowComponent(defaultProps)
        const instance: Gameboard = component.instance() as Gameboard
        instance.componentWillUnmount()
        expect(mockEvent.mock.calls[0][0]).toBe('resize')
      })
    })
  })

  describe('@methods', (): void => {
    describe('updateWindowDimensions', (): void => {
      it('should update the state with the window dimensions', (): void => {
        const component: ShallowWrapper = shallowComponent(defaultProps)
        component.setState({
          windowHeight: 40,
          windowWidth: 20,
        })
        const instance: Gameboard = component.instance() as Gameboard
        instance.updateWindowDimensions()
        expect(instance.state).toEqual({
          initialized: false,
          windowHeight: 768,
          windowWidth: 1024
        })
      })
    })

    describe('toggleTheater', (): void => {
      it('should remove map events', (): void => {
        const component: ShallowWrapper = shallowComponent(defaultProps)
        const instance: Gameboard = component.instance() as Gameboard
        instance.removeMapEvents = jest.fn()
        instance.toggleTheater()
        expect(instance.removeMapEvents).toHaveBeenCalled()
      })

      it('should toggle the theater on the store', (): void => {
        const component: ShallowWrapper = shallowComponent(defaultProps)
        const instance: Gameboard = component.instance() as Gameboard
        instance.removeMapEvents = jest.fn()
        instance.toggleTheater()
        expect(store.theater).toBe(URL_MAP_PAC)
      })
    })

    describe('getMapContext', (): void => {
      it('should  call switchTheaters and attachMapEvents if already initialized', () => {
        const component: ShallowWrapper = shallowComponent(defaultProps)
        const instance: Gameboard = component.instance() as Gameboard
        instance.attachMapEvents = jest.fn()
        
        // @ts-ignore
        instance.game = {
          switchTheaters: jest.fn()
        }
        component.setState({
          initialized: true
        })
        // @ts-ignore
        instance.getMapContext({}, {}, {});
        expect(instance.game.switchTheaters).toHaveBeenCalled();
        expect(instance.attachMapEvents).toHaveBeenCalled()
      })

      it('should create a gameInstance, attack Map and KeyboardEvents and setstate initialized if not already initialized', (): void => {
        const component: ShallowWrapper = shallowComponent(defaultProps)
        const instance: Gameboard = component.instance() as Gameboard
        instance.attachMapEvents = jest.fn()
        instance.attachKeyboardEvents = jest.fn()
        // @ts-ignore
        instance.getMapContext({}, {}, {});
        expect(instance.attachMapEvents).toHaveBeenCalled()
        expect(instance.attachKeyboardEvents).toHaveBeenCalled()
        expect(instance.state.initialized).toBe(true)          
      })

      it('should do nothing if the game is initialised but there is not game instance', (): void => {
        const component: ShallowWrapper = shallowComponent(defaultProps)
        const instance: Gameboard = component.instance() as Gameboard
        instance.attachMapEvents = jest.fn()
        instance.attachKeyboardEvents = jest.fn()
        // @ts-ignore
        instance.game = null
        component.setState({
          initialized: true
        })
        expect(instance.attachKeyboardEvents).not.toHaveBeenCalled()
        expect(instance.attachMapEvents).not.toHaveBeenCalled()
      })
    })

    describe('attachKeyboardEvents', (): void => {
    
      it('should attach keyboard events', (): void => {
        const mockEvent = jest.spyOn(window, 'addEventListener')
        const component = shallowComponent(defaultProps)
        const instance = component.instance() as Gameboard
        mockEvent.mockClear()
        instance.attachKeyboardEvents()
        expect(mockEvent.mock.calls[0][0]).toBe('keydown')
        expect(mockEvent.mock.calls[1][0]).toBe('keyup')
      })
    })

    describe('attachMapEvents', (): void => {
      let component: ShallowWrapper
      let instance: Gameboard
      beforeEach((): void => {
        component = shallowComponent(defaultProps)
        instance = component.instance() as Gameboard
        // @ts-ignore
        instance.mapCanvas = {
          addEventListener: jest.fn()
        }
        // @ts-ignore
        instance.game = {
          currentMap: {
            
            // @ts-ignore
            mapCanvas: null,
            onDoubleClick: {
              // @ts-ignore
              bind: jest.fn()
            },
            onMouseDown: {
              // @ts-ignore
              bind: jest.fn()
            },
            onMouseMove: {
              // @ts-ignore
              bind: jest.fn()
            },     
            onMouseUp: {
              // @ts-ignore
              bind: jest.fn()
            },
            onScroll: {
              // @ts-ignore
              bind: jest.fn()
            }
          }
        }
        // @ts-ignore
        instance.mainMapContainer = {
          // @ts-ignore
          current: {
            addEventListener: jest.fn()
          }
        }
      })

      it('should add eventListeners to mapCanvas', () => {
        instance.attachMapEvents()
        expect(instance.mapCanvas.addEventListener).toHaveBeenCalledWith('mousemove', undefined, false)
        expect(instance.mapCanvas.addEventListener).toHaveBeenCalledWith('mousedown', undefined, false)
        expect(instance.mapCanvas.addEventListener).toHaveBeenCalledWith('mouseup', undefined, false)
        expect(instance.mapCanvas.addEventListener).toHaveBeenCalledWith('dblclick', undefined, false)
      })

      it('should add scroll listener to mapcontainer if it is the current', () => {
        instance.attachMapEvents()
        // @ts-ignore
        expect(instance.mainMapContainer.current.addEventListener)
          .toHaveBeenCalledWith('scroll', undefined, true)
      })
    })

    describe('removeMapEvents', (): void => {
      let component: ShallowWrapper
      let instance: Gameboard
      beforeEach((): void => {
        component = shallowComponent(defaultProps)
        instance = component.instance() as Gameboard
        // @ts-ignore
        instance.mapCanvas = {
          removeEventListener: jest.fn()
        }
        // @ts-ignore
        instance.game = {
          currentMap: {
            // @ts-ignore
            mapCanvas: null
          }
        }
        // @ts-ignore
        instance.mainMapContainer = {
          // @ts-ignore
          current: {
            removeEventListener: jest.fn()
          }
        }
      })

      it('should remove eventListeners from mapCanvas', () => {
        instance.removeMapEvents()
        expect(instance.mapCanvas.removeEventListener).toHaveBeenCalledWith('mousemove', undefined, false)
        expect(instance.mapCanvas.removeEventListener).toHaveBeenCalledWith('mousedown', undefined, false)
        expect(instance.mapCanvas.removeEventListener).toHaveBeenCalledWith('mouseup', undefined, false)
        expect(instance.mapCanvas.removeEventListener).toHaveBeenCalledWith('dblclick', undefined, false)
      })

      it('should remove scroll listener from mapcontainer if it is the current', () => {
        instance.removeMapEvents()
        // @ts-ignore
        expect(instance.mainMapContainer.current.removeEventListener)
          .toHaveBeenCalledWith('scroll', undefined, true)
      })
    })
  })
})
