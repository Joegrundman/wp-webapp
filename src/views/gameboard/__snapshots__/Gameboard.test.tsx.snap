// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gameboard /> @render should render as expected in the default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameBoard
    store={
      GameStore {
        "currentActiveDialog": "",
        "showDialogPanel": false,
        "theater": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "GameStore@1.keys",
            "observers": Set {},
          },
          "name": "GameStore@1",
          "target": [Circular],
          "values": Map {
            "theater" => "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
            "showDialogPanel" => false,
            "currentActiveDialog" => "",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className={undefined}
        style={
          Object {
            "height": "678px",
            "width": "964px",
          }
        }
      >
        <MainMap
          getMapContext={[Function]}
          url="http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg"
        />
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <MainMap
          getMapContext={[Function]}
          url="http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg"
        />,
        "className": undefined,
        "style": Object {
          "height": "678px",
          "width": "964px",
        },
      },
      "ref": Object {
        "current": null,
      },
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "getMapContext": [Function],
          "url": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className={undefined}
          style={
            Object {
              "height": "678px",
              "width": "964px",
            }
          }
        >
          <MainMap
            getMapContext={[Function]}
            url="http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg"
          />
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MainMap
            getMapContext={[Function]}
            url="http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg"
          />,
          "className": undefined,
          "style": Object {
            "height": "678px",
            "width": "964px",
          },
        },
        "ref": Object {
          "current": null,
        },
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "getMapContext": [Function],
            "url": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<Gameboard /> @render should render as expected with canvas dimensions set on the state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameBoard
    store={
      GameStore {
        "currentActiveDialog": "",
        "showDialogPanel": false,
        "theater": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "GameStore@5.keys",
            "observers": Set {},
          },
          "name": "GameStore@5",
          "target": [Circular],
          "values": Map {
            "theater" => "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
            "showDialogPanel" => false,
            "currentActiveDialog" => "",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className={undefined}
        style={
          Object {
            "height": "210px",
            "width": "440px",
          }
        }
      >
        <MainMap
          getMapContext={[Function]}
          url="http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg"
        />
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <MainMap
          getMapContext={[Function]}
          url="http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg"
        />,
        "className": undefined,
        "style": Object {
          "height": "210px",
          "width": "440px",
        },
      },
      "ref": Object {
        "current": null,
      },
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "getMapContext": [Function],
          "url": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className={undefined}
          style={
            Object {
              "height": "210px",
              "width": "440px",
            }
          }
        >
          <MainMap
            getMapContext={[Function]}
            url="http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg"
          />
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MainMap
            getMapContext={[Function]}
            url="http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg"
          />,
          "className": undefined,
          "style": Object {
            "height": "210px",
            "width": "440px",
          },
        },
        "ref": Object {
          "current": null,
        },
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "getMapContext": [Function],
            "url": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<Gameboard /> @render should render as expected with different theater 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameBoard
    store={
      GameStore {
        "currentActiveDialog": "",
        "showDialogPanel": false,
        "theater": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179470/evnyvf9x1ehyxt3ksrkp.jpg",
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "GameStore@3.keys",
            "observers": Set {},
          },
          "name": "GameStore@3",
          "target": [Circular],
          "values": Map {
            "theater" => "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179470/evnyvf9x1ehyxt3ksrkp.jpg",
            "showDialogPanel" => false,
            "currentActiveDialog" => "",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className={undefined}
        style={
          Object {
            "height": "678px",
            "width": "964px",
          }
        }
      >
        <MainMap
          getMapContext={[Function]}
          url="http://res.cloudinary.com/druzhkwmt/image/upload/v1456179470/evnyvf9x1ehyxt3ksrkp.jpg"
        />
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <MainMap
          getMapContext={[Function]}
          url="http://res.cloudinary.com/druzhkwmt/image/upload/v1456179470/evnyvf9x1ehyxt3ksrkp.jpg"
        />,
        "className": undefined,
        "style": Object {
          "height": "678px",
          "width": "964px",
        },
      },
      "ref": Object {
        "current": null,
      },
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "getMapContext": [Function],
          "url": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179470/evnyvf9x1ehyxt3ksrkp.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className={undefined}
          style={
            Object {
              "height": "678px",
              "width": "964px",
            }
          }
        >
          <MainMap
            getMapContext={[Function]}
            url="http://res.cloudinary.com/druzhkwmt/image/upload/v1456179470/evnyvf9x1ehyxt3ksrkp.jpg"
          />
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MainMap
            getMapContext={[Function]}
            url="http://res.cloudinary.com/druzhkwmt/image/upload/v1456179470/evnyvf9x1ehyxt3ksrkp.jpg"
          />,
          "className": undefined,
          "style": Object {
            "height": "678px",
            "width": "964px",
          },
        },
        "ref": Object {
          "current": null,
        },
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "getMapContext": [Function],
            "url": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179470/evnyvf9x1ehyxt3ksrkp.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
